org.eclipse.jdt.launching/launching/org/eclipse/jdt/internal/launching/ClasspathShortener.java

```diff
@@ -294,8 +294,10 @@ private void shortenPathUsingClasspathArgumentFile(int modulePathArgumentIndex,
 		File file;
 		try {
 			String timeStamp = getLaunchTimeStamp();
-			File argFile = new File(processTempFilesDir, String.format(LAUNCH_TEMP_FILE_PREFIX + "%s" + option //$NON-NLS-1$
-					+ "-arg-%s.txt", getLaunchConfigurationName(), timeStamp)); //$NON-NLS-1$
+			String configName = (getLaunchConfigurationName().length() > 10) ? getLaunchConfigurationName().substring(0, 10)
```

Comment:
1) Please extract getLaunchConfigurationName() to a variable
2) Please extract 10 to a constant